Alliance.Common/Extensions/VOIP/Behaviors/VoiceDataManager.cs

```diff
@@ -0,0 +1,115 @@
+﻿using Alliance.Common.Extensions.VOIP.Models;
+using Alliance.Common.Extensions.VOIP.NetworkMessages;
+using NetworkMessages.FromServer;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using TaleWorlds.Library;
+using TaleWorlds.MountAndBlade;
+
+namespace Alliance.Common.Extensions.VOIP.Behaviors
+{
+    /// <summary>
+    /// Manage the list of Voice Data available to a player (the listener).
+    /// </summary>
+    public class VoiceDataManager
+    {
+        public NetworkCommunicator Listener { get; private set; }
+        private List<VoiceData> voiceDataList;
+
+        public VoiceDataManager(NetworkCommunicator listener)
+        {
+            Listener = listener;
+            voiceDataList = new List<VoiceData>();
+        }
```

Comment:
This seem to be used as single object. Maybe a Singleton would be good ?

Reply:
Maybe "Manager" isn't the right word but the server use one VoiceDataManager per client so we have several instances of it.

Reply:
I see. Maybe a comment on the class could help to understand then.
Or rename it into something like "ClientVoiceDataManager" in order to know that this is used by the server for each clients.